numpy/lib/arraysetops.py

```diff
@@ -315,18 +322,87 @@ def reshape_uniq(uniq):
         return uniq
 
     output = _unique1d(consolidated, return_index,
-                       return_inverse, return_counts, equal_nan=equal_nan)
+                       return_inverse, return_counts,
+                       equal_nan=equal_nan, kind=kind)
     output = (reshape_uniq(output[0]),) + output[1:]
     return _unpack_tuple(output)
 
 
 def _unique1d(ar, return_index=False, return_inverse=False,
-              return_counts=False, *, equal_nan=True):
+              return_counts=False, *, equal_nan=True, kind=None):
     """
     Find the unique elements of an array, ignoring shape.
     """
     ar = np.asanyarray(ar).flatten()
 
+    integer_array = np.issubdtype(ar.dtype, np.integer)
+
+    if kind not in {None, 'sort', 'table'}:
+        raise ValueError(
+            f"Invalid kind: '{kind}'. Please use None, 'sort' or 'table'.")
+    
+    use_table_method = (
+        integer_array
+        and kind in {'table', None}
+        and not return_index
+        and not return_inverse
+        and not isinstance(ar, np.ma.MaskedArray)
+        and not ar.size == 0
```

Comment:
Something about the final state of these exclusions should be mentioned in the docstring. It's not trivial that `kind='table'` actually means `kind='sort'` when you also ask for the index/reverse. But I understand you might end up implementing this for the new kind too.

Reply:
I made it so it throws an error if the conditions aren’t satisfied, rather than silently switching `kind`. The things left are to specify in the error why it is incompatible, and also describe in the docstring.

Reply:
Ah, sorry, I was reading from the bottom up and missed that. All good then.